commitizen/git.py

```diff
@@ -310,3 +310,10 @@ def _get_log_as_str_list(start: str | None, end: str, args: str) -> list[str]:
     if not c.out:
         return []
     return c.out.split(f"{delimiter}\n")
+
+
+def guess_default_branch() -> str:
+    c = cmd.run("git branch --format '%(refname:short)' --list master main")
```

Comment:
This makes a strong hypothesis that the reference branch can only be `master` or `main` so it won't work with `develop` which is common, or any other "exotic choice".

To handle this, I use `git branch -r | grep -Po "HEAD -> .+/\K.+$" | head -n 1` which works with any default branch as long as there is a remote with a default. The `head` is for the case where there are multiple remotes, but makes the hypothesis that they share the same default branch.

Reply:
I guess I'll have to try those commands 👀 Looks interesting!

I'm thinking of letting users set `default_git_remote` and `default_git_branch_name` (but not yet have time to impletemtn it 🥲)